### PR TITLE
- Netmera integration has been done, events firing structure has been set up for first screen.

### PR DESCRIPTION
**Pull request by yasirinitialcode**
## Related Issue
- Netmera integration.

ISSUE LINK: 

## Description
- Netmera integration has been done.
- User update request has been set up when signing in.
- Sign in event has been set up.
- Get started event has been set up.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?
- Tested with an emulator of Android 13, Redmi Note 7 (Android 10).

## Screenshots or Sample Output
<p float="left">
  

</p>

### CODE REVIEW

**Comment by yasirinitialcode** [comment]
Google-services.json file was added to previous PR.

File: app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        super.onCreate()
+
+        NMInitializer.initializeComponents(this)
+
+        AppInitializer.getInstance(this)
+
+        AppInitializer.getInstance(this).run {
+            initializeComponent(NMFCMProviderInitializer::class.java)
+            initializeComponent(NMHMSProviderInitializer::class.java)
+        }
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt]: Do we have to do this? @burak-initialcode Isn't this automatic?

File: app/src/main/java/com/netmera/netmerafintech/utils/AnalyticUtils.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        val event = GetStartedEvent()
+        event.setGetStarted(true)
+        Netmera.sendEvent(event)
+    }
+
+    fun updateNetmeraUser() {
+        val netmerauser = NMImpactFintechUser()
+        netmerauser.setUserId("burakaymaz@hotmail.com")
+        netmerauser.setName("Burak")
+        netmerauser.setType("black")
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/utils/AnalyticUtils.kt:22]: This is offensive! Whose idea was this :D

File: app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+        val configBuilder = NetmeraConfiguration.Builder()
+
+        configBuilder.apiKey(NETMERA_API_KEY)
+            .firebaseSenderId(FIREBASE_SENDER_ID)
```
Comment (ghost) [inline app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt]: I see you've added huawei libraries but didn't see huawei id here. Will you do that later?

**Review by ghost — APPROVED** [review]

File: app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt (hunk 3 above)
Comment (yasirinitialcode) [reply app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt]: I've added them to be able to integrate huawei later, but i think it was a mistake, i can add dependencies later easily. Will be removed in the next commit.

File: app/src/main/java/com/netmera/netmerafintech/utils/AnalyticUtils.kt (hunk 2 above)
Comment (yasirinitialcode) [reply app/src/main/java/com/netmera/netmerafintech/utils/AnalyticUtils.kt:22]: Burak Kaymaz instructed it.

Comment (ghost) [reply app/src/main/java/com/netmera/netmerafintech/utils/AnalyticUtils.kt:22]: I was joking. No problem :)

File: app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt (hunk 1 above)
Comment (bsoykal) [reply app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt]: We can remove all initializeComponent methods

File: app/build.gradle [hunk 4]
```diff
@@ -51,7 +51,12 @@ dependencies {
...
+
+    implementation 'com.netmera:nmcore:3.9.11'
+    implementation 'com.netmera:nmfcm:3.9.4'
 }
 
 apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-parcelize'
\ No newline at end of file
+apply plugin: 'kotlin-parcelize'
+
```
Comment (bsoykal) [inline app/build.gradle]: Please remove the unnecessary space.

File: app/src/main/java/com/netmera/netmerafintech/utils/Constants.kt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+package com.netmera.netmerafintech.utils
+
+const val FIREBASE_SENDER_ID = "151965488535"
+const val NETMERA_API_KEY = "RH9qH4AIng0xmH-dqrh3SYTSU3Kw3OrQpqAkyF-u6YOgwJJt_7n55WtPYT3ak-U_"
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/Constants.kt]: Move those to gradle instead.

File: app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import com.netmera.netmerafintech.utils.FIREBASE_SENDER_ID
+import com.netmera.netmerafintech.utils.NETMERA_API_KEY
+import com.netmera.nmfcm.NMFCMProviderInitializer
+
+class ImpactFintechApp: Application() {
+
+    override fun onCreate() {
+        super.onCreate()
+
+        NMInitializer.initializeComponents(this)
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt]: We don't need this one.

File: app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+        AppInitializer.getInstance(this)
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt]: No need.

File: app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+        AppInitializer.getInstance(this).run {
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/ImpactFintechApp.kt]: No need!

File: app/src/main/java/com/netmera/netmerafintech/utils/AnalyticUtils.kt [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+import android.app.Activity
+import android.os.Build
+import com.netmera.Netmera
+import com.netmera.netmerafintech.Netmera.Events.SignInEvent
+import com.netmera.netmerafintech.Netmera.Events.GetStartedEvent
+import com.netmera.netmerafintech.data.model.NMImpactFintechUser
+
+object AnalyticsUtil {
+
+    fun getNotificationPermission(activity: Activity) {
```
Comment (bsoykal) [inline app/src/main/java/com/netmera/netmerafintech/utils/AnalyticUtils.kt]: This has nothing to do with Analytics. Move this to something like AppUtils.

**Review by bsoykal — CHANGES_REQUESTED** [review]
Please find my inline comments and please add copyright information to all classes.

**Review by bsoykal — APPROVED** [review]
Very good

**Pull request merged**